---
title: "Survival Analysis of Post-Myocardial Infarction Patients"
author: "Alvein, Orr, Pham"
date: "5/22/2020"
output: pdf_document
---

```{r setup, include=FALSE, warning=FALSE, message=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_chunk$set(fig.height=4.5, fig.width=7)

library(readxl)
library(knitr)
library(tidyverse)
library(dplyr)
library(kableExtra)
library(survival)
library(survminer)
library(ggplot2)
library(VIM)
library(missForest)

df = data.frame(read_excel("df.xlsx"))
df[df=="?"] = " "
s.df = Surv(df$Survival,df$Status)
```


# Abstract  ####################################################################

### Background
The rates of myocardial infarction is becoming an increasing common occurence in the United States. Rapid development of medical technology and knowledge have led to an decline in myocardial infarction fatalities (Gu, et al 1999). However, there is much to be learned regarding the survival probabilities of patients following an infarction episode.Some studies have already examined the effects of externalities on the survival rates of these patients (Rimm, et al. 1995).

### Objectives
Our goal is to provide detailed survival statistics of patients during a post-myocardial infarction time period with specific concern addressed to age, ventricular activity, and physiological cardic state. Using these variables,  we aim to provide succinct information on the current state of the dataset as well as provide robust predictors for the future estimates of survival for future patients.

We aim to fit non-parametric (Kaplan-Meier) and parameters curves to describe the data as well as choose a regression model to be used for predictive surviability. 

### Methods
Data from 133 post-myocardial infarction patients measure the time in months until death in a one year monitoring period of follow-up. We use a combination of nonparametric (Kaplan-Meier) and parametric methods (Weilbull, Log-Normal, Log-Logistic, Cox PH) to determine estimates of survival among gender and physiological cardic state (contraction depth, muscular activity, anatomical status). We fit multiple distributions over the dataset to provide current-state information of the patient dataset. Then, we regress multiple models and use combination of Akaike Information  (AIC) statistics, logistic ratio tests, and residual analysis to determine model adequacy. 

### Results
Initial non-parametric Kaplan-Meier curve shows a median survival time of ~30 months for all age groups. We choose a Weibull regression fit (tenative) for predictive model as we have favorable AIC, ratio, and residual indicators out of all of our model.

### Conclusion
Thus, for predictive model we found the Weibull regression fit to be the most ideal candidate for modeling survivability for patient groups. Additionally, when examining the survival times for the Kaplan-Meier step curve, we see that the younger age groups do survive as well as their older counterparts. Given our limited sample size for that population, we recommend continued studies into external effects of the post-mycardial episode survival.




# Introduction  #################################################################################

Heart disease has become the leader cause of death among the US population among a majority of all racial and ethnic groups (Heron 2019). Myocardial infarctions are becoming largely common among U.S. populations. The number of myocardial infarctions are remarkedly increased from [start date] to [end date] by [x] amount (add citation). In 2015, approximately 23% of all fatalities in the United States was related to some degree of heart disease (cdc. cite please). Unsurprisingly, clinical studies have shown harmful symptoms in post-infarction survival patients. Our obtained dataset to examine the tangible difference in survability rates from the course of year following an infarction episode.

By applying survival analysis techniques to this data set, we seek to achieve improved understanding of the characteristics exhibited by patients in a one year post-infarction interval.We also propose a model to better predict the probability of a survival of patients based on these variable characteristics.


# Dataset ##################################################################################

We have obtained our data set from Kaggle. The data set contains 133 total patient observations and records 8 variables. Two patient survival times were not given; thus, we elected to remove those values to develop the most accurate portrayal of survival times. 

Since the time of myocardial infarction varies (depending if a patient joined the study prior to the starat), some patients were followed for less than a year. This provides a clear censoring and truncation provide. We will address this specifically in the following section.

It should be noted the the dataset have 40 missing values after the removal of the two patients. To develop estimates for the missing dated, we employed a k-nearest Neighbor algorithmic approach to predict the values for the missing values. With this in mind, our predictive and summary models will have less than ideal accuracy.

The reader may find a summary of tables and original dataset in the appendix of this paper.

### Imputation

In addition to the two rows that we removed, we further modified the dataset. The provided data contains 40 missing values that we chose to impute using the random forest algorithm methods in the missForest R package. Below is a summary of the missing data:

```{r missing, echo=FALSE, warning=FALSE, message=FALSE}

missing.data = aggr(df, col=c("paleturquoise3","sienna2"))
  title(main="Graphical Represenation of Missing Values in Original Dataset", line = 5)

kable(missing.data$missings, caption="Missing Values in Original Dataset") %>%
  kable_styling(position = "center")
```

The algorithmic process used here uses a modified k-nearest neighbot (KNN) approach. Using a training data set, the routines of the missForest algorithm predicts the missing values trained on the observed parts of the dataset (Stekhoven 2012). Refer to Stekhoven, et. al 2012 for more detail.

Following imputation, we verify the imputation accuracy using the normalized root mean squared error as an indicator of accuracy (NRMSE, Oba et al. (2003)). The general performance of our imputated dataset can be expressed by:

\[ NRMSE\ =\ \sqrt{\frac{mean\left(\left(X^{true}-X^{imp}\right)^2\right)}{var\left(X^{true}\right)}} \]

Where X is a matrix of our dataset. Our calculated NRMSE is as follows:

```{r imputation, echo=FALSE, warning=FALSE, message=FALSE}

df.m = prodNA(df, noNA = 0.2) #seed 10% of the missing values 
df.i = missForest(df.m)
kable(t(df.i$OOBerror), caption="Normal Root Mean Square Error") %>%
  kable_styling(position="center")

```

Being a random forest iterative process, each imputed dataset will be different from each other. Out of 500 iterations, we have an averaged NRMSE value of [] - that is our inputted values have an estimate []% deviation from true accuracy.

The full imputed dataset may be found in the appendix of this paper.

### Censoring

-Fixed study start time and fixed study end time.
-Patients can start late.
-Left censored: Some values, we are unsure how long ago was their myocardial infarction episode. We only have recorded data from the moment patients entered the study and when the study ends.
-Right censored: Additionally, some patients live far beyond the scope of the study. 




# Methodology ################################################################################

Here, we brief review the methodology and theory behind our analysis techniques for context.

## Non-Parametric Estimates

For our non-parametric estimates, we rely on Kaplan-Meier survival estimates.We conduct nonparametric Kaplan-Meier fit on our data over multiple strata. We first conduct a fit over all patients in regards to censoring. Then we work on gender groups, age, and general ventriculular condition. A survival function can be defined as:

[insert latex equation for survival function, with censored values]




## Parametric Estimates: Log-Normal Distribution



## Parametric Estimates: Log-Logistic Distribution



## Parametric Estimates: Weibull Distribution



## Hazard Estimates 



## Regression & Model Selection







# Results ##########################################################################################

## Exploratory Data Analysis




## Non-Parametric: Kaplan-Meier Survival Estimates


```{r km.all, fig.align='center', echo=FALSE}
km.all = survfit(s.df~1,type="kaplan-meier", data = df)
ggsurvplot(km.all, 
           palette = "#2E9FDF", 
           conf.int = TRUE, 
           title="Post-Myocardial Infarction Survival", 
           subtitle="All Groups",
           font.title=c(12,"bold.italic"),
           font.subtitle = c(10,"italic"),
           font.x = c(9, "bold.italic"),
           font.y = c(9, "bold.italic"),
           ylab="Surival Proportion", 
           xlab="Time to Death (Months)",
           surv.median.line = "hv",
           legend.title = "Groups",
           legend.labs = "All")
```

Here is the survival curve for all groups within our dataset. We see that a majority of our censored values have very short survival times. This is very intuitive given to our limited interval study time of a single year. We clearly see a median survival time of approximately 29 weeks.

```{r km.summary, fig.align='center', echo=FALSE, results = 'asis'}

kable(t(summary(km.all)$table)) %>%
  kable_styling(position = "center")
```

- Add stratified on age

- Add stratified on cardiac physiology

- Add pericardial effusion

- Add median and mean + 95% CI

- Add hazard plots



## Parametric: Log-Normal Model

```{r wb.plot, echo=FALSE}

```





## Parametric: Log-Logistic Model

```{r wb.plot, echo=FALSE}

```






## Parametric: Weibull Model

```{r wb.plot, echo=FALSE}

```







## Cox Proportional Hazard

```{r coxph.plot, echo=FALSE}

```

## Model Diagnostics






### AIC, BIC, and Confidence Intervals

```{r diagnostic.table, echo=FALSE}

```

### Residual Analysis/QQ Plot

```{r diagnostic.resid, echo=FALSE}

```

```{r diagnostic.qqplot, echo=FALSE}

```

# Discussion

# References

### Introduction

Gu K, Cowie CC, Harris MI. Diabetes and Decline in Heart Disease Mortality in US Adults. JAMA. 1999;281(14):1291–1297. doi:10.1001/jama.281.14.1291

Heron, M. Deaths: Leading causes for 2017 pdf icon[PDF – 3 M]. National Vital Statistics Reports;68(6). Accessed November 19, 2019.

Eric B. Rimm, Meir J. Stampfer, Edward Giovannucci, Alberto Ascherio, Donna Spiegelman, Graham A. Colditz, Walter C. Willett, Body Size and Fat Distribution as Predictors of Coronary Heart Disease among Middle-aged and Older US Men, American Journal of Epidemiology, Volume 141, Issue 12, 15 June 1995, Pages 1117–1127, https://doi.org/10.1093/oxfordjournals.aje.a117385


Kan, G., Visser, C., Kooler, J., & Dunning, A. (1986). Short and long term predictive value of wall motion score in acute myocardial infarction. British Heart Journal, 56, 422-427.

Fryar CD, Chen T-C, Li X. Prevalence of uncontrolled risk factors for cardiovascular disease: United States, 1999–2010 pdf icon[PDF-494K]. NCHS data brief, no. 103. Hyattsville, MD: National Center for Health Statistics; 2012. Accessed May 9, 2019.

### Methodology

Tableman, M., & Kim, J. S. (2004). Survival analysis using S: analysis of time-to-event data. Boca Raton, Florida: Chapman & Hall.

Salzberg, S. (1988). Exemplar-based learning: Theory and implementation (Technical Report TR-10-88). Harvard University, Center for Research in Computing Technology, Aiken Computation Laboratory (33 Oxford Street; Cambridge, MA 02138).

### Imputation

Oba S, Sato MA, Takemasa I, Monden M, Matsubara K, Ishii S. A Bayesian missing value estimation method for gene expression profile data. Bioinformatics. 2003;19(16):2088‐2096. doi:10.1093/bioinformatics/btg287

Daniel J. Stekhoven, Peter Bühlmann, MissForest—non-parametric missing value imputation for mixed-type data, Bioinformatics, Volume 28, Issue 1, 1 January 2012, Pages 112–118,

### Discussion

Ford, E. S., & Capewell, S. (2007). Coronary Heart Disease Mortality Among Young Adults in the U.S. From 1980 Through 2002. Journal of the American College of Cardiology, 50(22), 2128–2132. doi: 10.1016/j.jacc.2007.05.056

Andrikopoulos, G. K., Tzeis, S. E., Pipilis, A. G., Richter, D. J., Kappos, K. G., Stefanadis, C. I., … Chimonas, E. T. (2006). Younger age potentiates post myocardial infarction survival disadvantage of women. International Journal of Cardiology, 108(3), 320–325. doi: 10.1016/j.ijcard.2005.05.016

Sia, Y. T., Parker, T. G., Liu, P., Tsoporis, J. N., Adam, A., & Rouleau, J. L. (2002). Improved post-myocardial infarction survival with probucol in rats: Effects on left ventricular function, morphology, cardiac oxidative stress and cytokine expression. Journal of the American College of Cardiology, 39(1), 148–156. doi: 10.1016/s0735-1097(01)01709-0

\newpage

# Appendix 

## Dataset Variable Summary

```{r Dataset.summary, echo=FALSE, results='asis'}

df.sum = data.frame(read_excel("df.sum.xlsx"))

kable(df.sum, "latex", 
      booktabs = TRUE,
      longtable = TRUE,
      linesep = "\\addlinespace",
      caption = "Summary of Dataset Covariates") %>%
  kable_styling(latex_options = c("hold_positon","repeat_header"),
                full_width = TRUE)

```





## Original Dataset

```{r Dataset.actual, echo=FALSE, results='asis'}

kable(df, "latex", 
      booktabs = TRUE,
      longtable = TRUE,
      caption = "Original Dataset") %>%
  kable_styling(latex_options = c("hold_positon","repeat_header"),
                full_width = TRUE)
```





## Imputed Dataset

```{r imp.table, echo=FALSE, results='asis', warning=FALSE, message=FALSE}

round_df <- function(x, digits) {
    # round all numeric variables
    # x: data frame 
    # digits: number of digits to round
    numeric_columns <- sapply(x, mode) == 'numeric'
    x[numeric_columns] <-  round(x[numeric_columns], digits)
    x}

df.new = round_df(df.i$ximp,2)

kable(df.new, "latex", 
      booktabs = TRUE,
      longtable = TRUE,
      caption = "Imputed Dataset") %>%
  kable_styling(latex_options = c("hold_positon","repeat_header"),
                full_width = TRUE)

```

\newpage

## R Code

```{r ref.label=knitr::all_labels(), echo = T, eval = F}
```


\newpage
