---
title: "Survival Analysis of Post-Myocardial Infarction Patients"
author: "Alvein, Orr, Pham"
date: "5/22/2020"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_chunk$set(fig.height=4.5, fig.width=7)

library(readxl)
library(knitr)
library(tidyverse)
library(dplyr)
library(kableExtra)
library(survival)
library(survminer)
library(ggplot2)
library(arsenal)
library(VIM)
library(missForest)

df = data.frame(read_excel("df.xlsx"))
df[df=="?"] = " "
s.df = Surv(df$Survival,df$Status)
```
# Abstract

### Background
The rates of myocardial infarction is becoming an increasing common occurence in the United States. As medical knowledge and techniques improve to meet need of infarction episode, so does the need to understand the survivbility of patients who have survived such episodes.

### Objectives
Our goal is to provide detailed survival statistics of post-myocardial infarction patients as well as provide an accurate regression model to best prediction of survival outcomes of a single year following an infarction episode.

### Methods
Data from 133 post-myocardial infarction patients measure the time in months until death in a one year monitoring period of follow-up. We use a combination of nonparametric (Kaplan-Meier) and parametric methods (Weibull/Cox PH) to determine estimates of survival among gender and myocardial strata (contraction depth, muscular activity, ). We consider a slew of statistical and graphical results before determining the most appropriate method of modeling.

### Results
Out of all of our methods, we have determined that [] is the most appropriate model for prediction of patient survival. We have AIC values of. We have BIC values. Thus, this model is the best.

### Conclusion
[summary statistics]
[review of our model + specific surval rates]


# Introduction

Heart disease has become the leader cause of death among the US population among a majority of all racial and ethnic groups (Heron 2019). Myocardial infarctions are becoming largely common among U.S. populations. The number of myocardial infarctions are remarkedly increased from [start date] to [end date] by [x] amount (add citation). In 2015, approximately 23% of all fatalities in the United States was related to some degree of heart disease (cdc. cite please). Unsurprisingly, clinical studies have shown harmful symptoms in post-infarction survival patients. Our obtained dataset to examine the tangible difference in survability rates from the course of year following an infarction episode.

By applying survival analysis techniques to this data set, we seek to achieve improved understanding of the characteristics exhibited by patients in a one year post-infarction interval.We also propose a model to better predict the probability of a survival of patients based on these variable characteristics.


## Dataset
We have obtained our data set from Kaggle. The data set contains 133 total patient observations and records 8 variables. Two of the patients as survival times were not given; thus, we elected to remove those values to develop the most accurate portrayal of survival times. 

Since the time of infartion varies, some patients were followed for less than a year. This provides a clear censoring and truncation provide. We will address this concern in detail in the later in this section. It should be noted that we have a slew of missing values. Since a single patient has shown as missing, we have opted to impute the values for this data row. With this in mind, our predictive and summary models will have less than ideal accuracy.

The reader may find a summary of tables and original dataset in the appendix of this paper.


# Methodology

## Imputation

In addition to the two rows that we removed, we further modified the dataset. The provided data contains 40 missing values that we chose to impute using the random forest algorithm methods in the missForest R package. Below is a summary of the missing data:

```{r missing, echo=FALSE}

missing.data = aggr(df,
                    col=c("paleturquoise3","sienna2"),
                    sortVars=TRUE,)
missing.data
```

The algorithmic process used here uses a modified k-nearest neighbot (KNN) approach. sing a training data set, the routines of the missForest algorithm predicts the missing values trained on the observed parts of the dataset (Stekhoven 2012). Refer to Stekhoven, et. al 2012 for more detail.

Following imputation, we verify the imputation accuracy using the normalized root mean squared error as an indicator of accuracy (NRMSE, Oba et al. (2003)). The general performance of our imputated dataset can be expressed by:

\[ NRMSE = \sqrt[2]{\frac{mean((\text{X}^true - \text{X}^imputed)^2)}{var(\text{X}^true)} \]

Our calculated NRMSE is as follows:

```{r imputation, echo=FALSE}

df.m = prodNA(df, noNA = 0.1) #seed 10% of the missing values 
df.i = missForest(df.m)
kable(t(df.i$OOBerror))

```

The full imputed dataset may be found in the appendix of this paper.

## Censoring

Fixed study start time and fixed study end time.

Patients can start late.

Left censored: Some values, we are unsure how long ago was their myocardial infarction episode. We only have recorded data from the moment patients entered the study and when the study ends.

Right censored: Additionally, some patients live far beyond the scope of the study. 




## Summary statistics
Over the course of the study, there is a 
## Kaplan-Meier
We conduct nonparametric Kaplan-Meier fit on our data over multiple strata. We first conduct a fit over all patients in regards to censoring. Then we work on gender groups, age, and general ventriculular condition. A survival function can be defined as:

[insert latex equation for survival function, with censored values]

## Weibull Fits (+Cox PH)

## Log-Normal



# Results
## Table of Summary Statistics

```{r stats.summary, echo=FALSE}

```

## KM Curve & Hazard

### Survival Curves
#### KM Overall
```{r km.all, fig.align='center', echo=FALSE}
km.all = survfit(s.df~1,type="kaplan-meier", data = df)
ggsurvplot(km.all, 
           palette = "#2E9FDF", 
           conf.int = TRUE, 
           title="Post-Myocardial Infarction Survival", 
           subtitle="All Groups",
           font.title=c(12,"bold.italic"),
           font.subtitle = c(10,"italic"),
           font.x = c(9, "bold.italic"),
           font.y = c(9, "bold.italic"),
           ylab="Surival Proportion", 
           xlab="Time to Death (Months)",
           surv.median.line = "hv",
           legend.title = "Groups",
           legend.labs = "All")
```

Here is the survival curve for all groups within our dataset. We see that a majority of our censored values have very short survival times. This is very intuitive given to our limited interval study time of a single year. We clearly see a median survival time of approximately 29 weeks.

```{r km.summary, fig.align='center', echo=FALSE, results = 'asis'}

kable(t(summary(km.all)$table))
```






### Hazard Plots

## Weibull Curve

```{r wb.plot, echo=FALSE}

```

## Cox Proportional Hazard

```{r coxph.plot, echo=FALSE}

```

## Model Diagnostics

### AIC, BIC, and Confidence Intervals

```{r diagnostic.table, echo=FALSE}

```

### Residual Analysis/QQ Plot

```{r diagnostic.resid, echo=FALSE}

```

```{r diagnostic.qqplot, echo=FALSE}

```

# Discussion

# References

Heron, M. Deaths: Leading causes for 2017 pdf icon[PDF – 3 M]. National Vital Statistics Reports;68(6). Accessed November 19, 2019.

Oba S, Sato MA, Takemasa I, Monden M, Matsubara K, Ishii S. A Bayesian missing value estimation method for gene expression profile data. Bioinformatics. 2003;19(16):2088‐2096. doi:10.1093/bioinformatics/btg287

Daniel J. Stekhoven, Peter Bühlmann, MissForest—non-parametric missing value imputation for mixed-type data, Bioinformatics, Volume 28, Issue 1, 1 January 2012, Pages 112–118,

Salzberg, S. (1988). Exemplar-based learning: Theory and implementation (Technical Report TR-10-88). Harvard University, Center for Research in Computing Technology, Aiken Computation Laboratory (33 Oxford Street; Cambridge, MA 02138).

Kan, G., Visser, C., Kooler, J., & Dunning, A. (1986). Short and long term predictive value of wall motion score in acute myocardial infarction. British Heart Journal, 56, 422-427.

Fryar CD, Chen T-C, Li X. Prevalence of uncontrolled risk factors for cardiovascular disease: United States, 1999–2010 pdf icon[PDF-494K]. NCHS data brief, no. 103. Hyattsville, MD: National Center for Health Statistics; 2012. Accessed May 9, 2019.


# Appendix

## Dataset Variable Summary

```{r Dataset.summary, echo=FALSE, results='asis'}

df.sum = data.frame(read_excel("df.sum.xlsx"))

df.sum %>%
  kable() %>%
  kable_styling(full_width =TRUE)

```

## Original Dataset

```{r Dataset.actual, echo=FALSE, results='asis'}

kable(df, caption = "Dataset")
```

## Imputed Dataset

```{r imp.table, echo=FALSE, results='asis'}

df.m = prodNA(df, noNA = 0.1) #seed 10% of the missing values 
df.i = missForest(df.m)
round_df <- function(x, digits) {
    # round all numeric variables
    # x: data frame 
    # digits: number of digits to round
    numeric_columns <- sapply(x, mode) == 'numeric'
    x[numeric_columns] <-  round(x[numeric_columns], digits)
    x}

df.new = round_df(df.i$ximp,2)
kable(df.new)

```


## R Code

